lims 2123, Fix cicd

GitEdvard opened the pull request:
Purpose: 
The CI/CD stopped to work under travis. 

Solution:
I switched from travis to github actions instead. 

Validation:
I have verified that clarity-ext is installed without errors (in logs). I have also pushed a failing unit test and verified that the CI/CD fails in that case.

GitEdvard committed:
Add example github action

GitEdvard committed:
Install python

GitEdvard committed:
Install clarity ext

GitEdvard committed:
Run unit tests

GitEdvard committed:
Add failing unit test

GitEdvard committed:
Clean up example code, restore test

GitEdvard committed:
Trigger action on both push and pull request

matrulda commented on .github/workflows/github-actions-demo.yml:
```diff
@@ -0,0 +1,24 @@
+name: GitHub Actions Demo
```
It would be nice to rename this file, workflow and job name to something similar to "Run clarity-ext unit tests"

GitEdvard replied:
Yes, when I arrived at home yesterday, I realized that I forgot to rename the demo file :)

matrulda commented on .github/workflows/github-actions-demo.yml:
```diff
@@ -0,0 +1,24 @@
+name: GitHub Actions Demo
+on: [push, pull_request]
+jobs:
+  Explore-GitHub-Actions:
```
See above comment.

matrulda commented on .github/workflows/github-actions-demo.yml:
```diff
@@ -0,0 +1,24 @@
+on: [push, pull_request]
+jobs:
+  Explore-GitHub-Actions:
+    runs-on: ubuntu-latest
```
There's a risk something could break when just using latest, so I would suggest using a specific version, for example ubuntu-20.04

GitEdvard replied:
:+1:

matrulda commented on .github/workflows/github-actions-demo.yml:
```diff
@@ -0,0 +1,24 @@
+
+    steps:
+      - name: Set up Python ${{ matrix.python-version }}
+        uses: actions/setup-python@v1
```
I happen to know there's a v3 for this release action, perhaps use that instead? When implementing something new I think it's worth trying to use the most recent versions :)

GitEdvard replied:
:+1:

matrulda reviewed (changes requested):
Looks good! Left some minor comments. Not really necessary, but I think it would be nice to change the demo names at least. Let me know if you feel differently. 

I'm not sure while one of the jobs still are running? But hopefully it's just a temporary hickup.

GitEdvard committed:
Remove trigger on pull request

I think trigger on push is enough, and there was a job that hanged for
pull request, so I skip it.

GitEdvard committed:
Give action a relevant name

GitEdvard committed:
Run unit tests in a specific ubuntu version

GitEdvard committed:
Change to setup python v3

GitEdvard commented:
Thank you for the feedback. I have now adjusted accordingly.

matrulda reviewed (approved):
Nice, good job! Great to get rid of travis :smile:
